### PR TITLE
Take snapshots of downloads.v1.json

### PR DESCRIPTION
**Pull request by scottbommarito**
https://github.com/NuGet/Engineering/issues/2574

Statistics DB doesn't have the data for older than 6 weeks, so we've decided to start taking snapshots of `downloads.v1.json` to get around this.

[Azure does not restrict the number of snapshots of a file](https://docs.microsoft.com/en-us/rest/api/storageservices/creating-a-snapshot-of-a-blob), so we should be able to keep daily download counts forever using this method.

### CODE REVIEW

**Comment by loic-sharma** [comment]
Azure Search already does this: https://github.com/NuGet/NuGet.Services.Metadata/pull/597

Should we use that instead of snapshots?

**Comment by scottbommarito** [comment]
@loic-sharma - @joelverhagen and I have talked offline and he thinks this is a better solution.

**Comment by loic-sharma** [comment]
Could you summarize that discussion - is it because Azure Search stores the downloads in a different format?

**Comment by joelverhagen** [comment]
> Could you summarize that discussion - is it because Azure Search stores the downloads in a different format?

Azure Search version has a copy per region across multiple containers (-005, -007, ...). This is an undesirable side effect for coupling this with Azure Search storage containers.

The format is pretty much irrelevant. We have code to build and parse both formats so whoever is reading or writing can just use whatever. Yeah, downloads.v2.json is more pretty (if I do say so myself) but it needs further cleaning before it's to be used more broadly for reasons discussed offline so it's not 👌 quite yet.

This is the producer code so a snapshot here is in a "push" fashion instead of a "pull" fashion. This makes more sense since the data is only going to change when this stats job runs. This is not a huge point since Azure Search only uploads a snapshot when there is a change from the previous. A real "push" model would be triggered by log delivery but that's for a future age, future stats pipeline.

After this is merged I will try to create an issue to delete the Azure Search downloads.v2.json snapshotting. We don't need more than one history. We will keep the owner delta snapshotting in Azure Search since there is no counterpart and it requires a bit more work anyway (the snapshot is a "list of package IDs that had changes" for GDPR reasons instead of a "list of a package IDs and their associated data" so you need more context).

The only drawback of this approach is that a snapshot is made even if there is not a change. This is a non-issue since PROD downloads always go up per iteration (currently 3 hours). DEV and INT should have this also due to monitoring downloads but I'm less sure here and like who cares anyways 🤷‍♀.

Arguably N blobs with different names are more consumable than blob snapshots but any benefit of this style in Azure Search is probably lost (and then some) since you have to pick a region (arbitrary) and scan N containers to get the full history. So it's a bit of a wash. 

Also, this code is smaller 💯

**Comment by scottbommarito** [comment]
Thanks for the approvals!

I will be deploying this to DEV and testing it before I merge.

**Comment by scottbommarito** [comment]
DEV looks good!
